src/docfx/build/toc/TableOfContentsMap.cs

```diff
@@ -48,57 +50,40 @@ public string FindTocRelativePath(Document file)
         /// "near" means less subdirectory count
         /// when subdirectory counts are same, "near" means less parent directory count
         /// e.g. "../../a/TOC.md" is nearer than "b/c/TOC.md".
-        /// when the file is not referenced, return only toc in the same folder or parents folder.
-        /// i.e. relativePath only contains "..".
+        /// when the file is not referenced, return only toc in the same or higher folder level.
         /// </summary>
         public Document GetNearestToc(Document file)
         {
-            // fallback to all tocs if no toc files reference this file
-            var hasReferencedToc = _documentToTocs.TryGetValue(file, out var referencedTocFiles);
-            var filteredTocFiles = hasReferencedToc ? (IEnumerable<Document>)referencedTocFiles : _tocs;
-
-            var nearestToc = (Document)null;
-            var nearestSubDirCount = 0;
-            var nearestParentDirCount = 0;
-            foreach (var toc in filteredTocFiles)
-            {
-                var relativePath = PathUtility.GetRelativePathToFile(file.SitePath, toc.SitePath);
-                var (subDirCount, parentDirCount) = GetDirectoryCount(relativePath);
-                if (!hasReferencedToc && subDirCount != 0)
-                {
-                    continue;
-                }
-                var distance = Compare(nearestSubDirCount, nearestParentDirCount, subDirCount, parentDirCount);
-                if (nearestToc == null || distance > 0 ||
-                    (distance == 0 && string.Compare(nearestToc.SitePath, toc.SitePath, PathUtility.PathComparison) > 0))
-                {
-                    nearestToc = toc;
-                    nearestSubDirCount = subDirCount;
-                    nearestParentDirCount = parentDirCount;
-                }
-            }
+            var hasReferencedTocs = false;
+            var filteredTocs = (hasReferencedTocs = _documentToTocs.TryGetValue(file, out var referencedTocFiles)) ? referencedTocFiles : _tocs;
 
-            return nearestToc;
-        }
+            var fileNames = Path.GetFileNameWithoutExtension(file.SitePath).Split(new[] { '-', '#', '_' })
```

Comment:
>Split [](start = 76, length = 5)

file name can have space

Reply:
do we need consider that case like AzureWebapp.md?

---
In reply to: [220782407](https://github.com/dotnet/docfx/pull/3450#discussion_r220782407) [](ancestors = 220782407)